Oleksandr_Mykhailovskyi/4/game_utilities.py

```diff
@@ -0,0 +1,287 @@
+}
+
+
+def format_cell(mstr, cell_index):
+    """
+    Gets str reprs regarding player pos.
+
+    Args:
+        mstr (str): str repr of cell.
+        cell_index ([int, int]): cell index
+    Returns:
+        str - repr of cell regarding player pos.
+    """
+    mrepr = ""
+    if cell_index[0] == position[0] and cell_index[1] == position[1]:
+        mrepr = player_repr
+    elif cell_index[0] > position[0] + view_depth[0] or cell_index[1] > position[1] + view_depth[1]:
+        mrepr = fog_repr
+    elif cell_index[0] < position[0] - view_depth[0] or cell_index[1] < position[1] - view_depth[1]:
+        mrepr = fog_repr
+    elif mstr == map_elements[1][1] or mstr == map_elements[2][1]:
+        mrepr = unknown_repr
+    else:
+        mrepr = mstr
+    return mrepr
+
+
+def format_output(game_map):
+    """
+    Shows map.
+
+    Args:
+        game_map ([[str...]...]): game map.
+    """
+
+    print(position)
+
+    printed = str()
+    for index_i, i in enumerate(game_map):
+        for index_j, j in enumerate(i):
+            printed += format_cell(j, [index_i, index_j])
+        printed += "\n"
+    print(printed)
+
+
+def get_map_cell(position, game_map):
+    """
+    Little hack ot check cell's around player
+    when he's on the border.
+
+    Args:
+        position ((int,int)): position to check.
+        game_map ([[str...]...]): game map generated by game_generator module.
+
+    Returns:
+        cell content if position is valid.
+        map_cells_repr[0] - for non-valid position.
+    """
+    size = len(game_map)
+    if position[0] >= size or position[0] < 0 or \
+            position[1] >= size or position[0] < 0:
+
+        return map_cells_repr[0]
+
+    return game_map[position[0]][position[1]]
+
+
+def get_action():
+    """
+    Returns:
+        str - Action mapped on input.
+        If no action / no such action - None
+    """
+
+    input_action = input("Your turn\n")
+    for key in action_dict.keys():
+        if input_action == action_dict[key]:
+            return key
+    return None
+
+
+def check_position(position, game_map):
+    """
+    Checks whether position is acceptable.
+
+    Args:
+        position ((int,int)): position to check.
+        game_map ([[str...]...]): game map generated by game_generator module.
+    Returns:
+        True if acceptable.
+        False otherwise.
+    """
+    size = len(game_map)
+    if position[0] >= size or position[1] >= size:
+        return False
+    elif position[0] < 0 or position[1] < 0:
+        return False
+    return True
+
+
+def move_player(action, position, game_map):
+    """
+    Moves player on the map.
+
+    Args:
+        action (str): action from action dict.
+        position ((int,int)): position to check.
+        game_map ([[str...]...]): game map generated by game_generator module.
+    Returns:
+        [int, int] - list, new position.
+    """
+    oldPosition = list(position)
+    if action == actions_list[0]:
```

Comment:
Yes, but for each action I have to do a different thing, but then I would need to store some container of lambdas to make a bijection between index and lambda. I thought it would be an overkill since I only have like 4-5 actions(up,down,left,right,save)